src/nlsolve/newton.jl

```diff
@@ -173,9 +173,12 @@ end
     reltol = eps(eltype(dz))
   end
 
-  linsolve(vec(dz), W, b, iter == 1 && new_W;
-           Pl=DiffEqBase.ScaleVector(weight, true),
-           Pr=DiffEqBase.ScaleVector(weight, false), reltol=reltol)
+  if iter == 1 && new_W
+    linsolve = set_A(linsolve,W)
+  end
+  linsolve = set_b(linsolve,b)
+  linres = solve(linsolve,weights=weight,reltol=reltol)
```

Comment:
@vpuri3 we need some way to update the weights of the default preconditioners very easily.

Reply:
this should do

`linsolve = sete_prec(linsolve,scaling_preconditioner(weight)...)`

Reply:
would this be a reasonable use case for this interface?

https://github.com/SciML/LinearSolve.jl/pull/21#issuecomment-978078734

i think it would be a cleaner to call a single `solve(prob, cache)` for reusing cache for solving multiple problems.

```julia
function solve(prob, cache; scale=nothing, Pl=nothing, Pr=nothing, kwargs...)
    prob.A !== cache.A && cache = set_A(cache, prob.A)
    prob.b !== cache.b && cache = set_b(cache, prob.b)
    prob.u !== cache.u && cache = set_u(cache, prob.u0)
    scale  !== nothing && cache = set_prec(cache,scaling_preconditioner(weight)...)
    # similar handling for other kwargs
    return solve(cache, kwargs...)
end

prob = LinearProblem(A,b;u0=u, kwargs...)
solve(prob,cache; kwargs...)
```